qadence/ml_tools/train_grad.py

```diff
@@ -177,16 +198,32 @@ def loss_fn(model: torch.nn.Module, data: torch.Tensor) -> tuple[torch.Tensor, d
                 if iteration % config.write_every == 0:
                     write_tensorboard(writer, loss, metrics, iteration)
 
+                if perform_val:
+                    if iteration % config.val_every == 0:
+                        # TODO: It may be desired that the entire validation set be used
```

Comment:
This is a question that I would like to know someone's thoughts on. I believe it's more useful to iterate over the entire validation data whenever a validation check needs to be performed. Currently, this check is only done for a single batch, and that batch also changes at different iterations when a validation check is performed.

That can be done by a for/while loop to loop over the entire validation data with the below logic:

```py
def validate():
    val_loss = 0.0
    for data in val_dataloader:
        data_to_device = data_to_device(data, device=device, dtype=data_dtype)
        loss, _ = loss_fn(model, data_to_device)
        val_loss += loss
    return val_loss
```

Reply:
i see your point. lets just validate on batches for now